source/common/http/conn_pool_grid.h

```diff
@@ -206,9 +212,9 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   // that specifies HTTP/3 and HTTP/3 is not broken.
   bool shouldAttemptHttp3();
 
-  // Creates the next pool in the priority list, or nullptr if all pools have been created.
-  // TODO(alyssawilk) replace this now we have explicit pools.
-  virtual ConnectionPool::Instance* createNextPool();
+  // Creates the requested pool
+  virtual ConnectionPool::Instance* getOrCreateHttp3Pool();
```

Comment:
Looking at the tests, it seems that we really only want to override the "create" part of this method (not the "get" part or the "setupPool" parts, etc). So how 'bout we cook up `createHttp3Pool()` and `createHttp2Pool()` and make them virtual. Then they can be called from the non-virutal `getOrCreate()`. This would simplify the mocking. (Alternatively we could pass two `absl::AnyInvokable`s in the constructor and call them without using any virtual functions.

Reply:
done.